File: code/stable/glassbr/src/cpp/InputFormat.cpp
#include "InputFormat.hpp"

#include <algorithm>
#include <iostream>
#include <fstream>
#include <iterator>
#include <string>
#include <math.h>
#include <sstream>
#include <limits>
#include <vector>

using std::string;
using std::vector;
using std::ifstream;
using std::ofstream;

#include "InputParameters.hpp"

void get_input(string filename, InputParameters &inParams) {
    ofstream outfile;
    outfile.open("log.txt", std::fstream::app);
    outfile << "function get_input called with inputs: {" << std::endl;
    outfile << "  filename = ";
    outfile << filename;
    outfile << ", " << std::endl;
    outfile << "  inParams = ";
    outfile << "Instance of InputParameters object" << std::endl;
    outfile << "  }" << std::endl;
    outfile.close();
    
    ifstream infile;
    infile.open(filename, std::fstream::in);
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> inParams.a;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'inParams.a' assigned to ";
    outfile << inParams.a;
    outfile << " in module InputFormat" << std::endl;
    outfile.close();
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> inParams.b;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'inParams.b' assigned to ";
    outfile << inParams.b;
    outfile << " in module InputFormat" << std::endl;
    outfile.close();
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> inParams.w;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'inParams.w' assigned to ";
    outfile << inParams.w;
    outfile << " in module InputFormat" << std::endl;
    outfile.close();
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> inParams.P_btol;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'inParams.P_btol' assigned to ";
    outfile << inParams.P_btol;
    outfile << " in module InputFormat" << std::endl;
    outfile.close();
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> inParams.TNT;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'inParams.TNT' assigned to ";
    outfile << inParams.TNT;
    outfile << " in module InputFormat" << std::endl;
    outfile.close();
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> inParams.g;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'inParams.g' assigned to ";
    outfile << inParams.g;
    outfile << " in module InputFormat" << std::endl;
    outfile.close();
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> inParams.t;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'inParams.t' assigned to ";
    outfile << inParams.t;
    outfile << " in module InputFormat" << std::endl;
    outfile.close();
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> inParams.SD_x;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'inParams.SD_x' assigned to ";
    outfile << inParams.SD_x;
    outfile << " in module InputFormat" << std::endl;
    outfile.close();
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> inParams.SD_y;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'inParams.SD_y' assigned to ";
    outfile << inParams.SD_y;
    outfile << " in module InputFormat" << std::endl;
    outfile.close();
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> inParams.SD_z;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    outfile.open("log.txt", std::fstream::app);
    outfile << "var 'inParams.SD_z' assigned to ";
    outfile << inParams.SD_z;
    outfile << " in module InputFormat" << std::endl;
    outfile.close();
    infile.close();
}

File: code/stable/GlassBR/src/java/GlassBR/InputFormat.java
package GlassBR;

/** \file InputFormat.java
    \author Nikitha Krithnan and W. Spencer Smith
    \brief Provides the function for reading inputs
*/
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.Scanner;

public class InputFormat {
    
    /** \brief Reads input from a file with the given file name
        \param filename name of the input file
        \param inParams structure holding the input values
    */
    public static void get_input(String filename, InputParameters inParams) throws FileNotFoundException, IOException {
        PrintWriter outfile;
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.println("function get_input called with inputs: {");
        outfile.print("  filename = ");
        outfile.print(filename);
        outfile.println(", ");
        outfile.print("  inParams = ");
        outfile.println("Instance of InputParameters object");
        outfile.println("  }");
        outfile.close();
        
        Scanner infile;
        infile = new Scanner(new File(filename));
        infile.nextLine();
        inParams.a = Double.parseDouble(infile.nextLine());
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'inParams.a' assigned ");
        outfile.print(inParams.a);
        outfile.println(" in module InputFormat");
        outfile.close();
        infile.nextLine();
        inParams.b = Double.parseDouble(infile.nextLine());
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'inParams.b' assigned ");
        outfile.print(inParams.b);
        outfile.println(" in module InputFormat");
        outfile.close();
        infile.nextLine();
        inParams.w = Double.parseDouble(infile.nextLine());
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'inParams.w' assigned ");
        outfile.print(inParams.w);
        outfile.println(" in module InputFormat");
        outfile.close();
        infile.nextLine();
        inParams.P_btol = Double.parseDouble(infile.nextLine());
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'inParams.P_btol' assigned ");
        outfile.print(inParams.P_btol);
        outfile.println(" in module InputFormat");
        outfile.close();
        infile.nextLine();
        inParams.TNT = Double.parseDouble(infile.nextLine());
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'inParams.TNT' assigned ");
        outfile.print(inParams.TNT);
        outfile.println(" in module InputFormat");
        outfile.close();
        infile.nextLine();
        inParams.g = infile.nextLine();
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'inParams.g' assigned ");
        outfile.print(inParams.g);
        outfile.println(" in module InputFormat");
        outfile.close();
        infile.nextLine();
        inParams.t = Double.parseDouble(infile.nextLine());
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'inParams.t' assigned ");
        outfile.print(inParams.t);
        outfile.println(" in module InputFormat");
        outfile.close();
        infile.nextLine();
        inParams.SD_x = Double.parseDouble(infile.nextLine());
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'inParams.SD_x' assigned ");
        outfile.print(inParams.SD_x);
        outfile.println(" in module InputFormat");
        outfile.close();
        infile.nextLine();
        inParams.SD_y = Double.parseDouble(infile.nextLine());
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'inParams.SD_y' assigned ");
        outfile.print(inParams.SD_y);
        outfile.println(" in module InputFormat");
        outfile.close();
        infile.nextLine();
        inParams.SD_z = Double.parseDouble(infile.nextLine());
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.print("var 'inParams.SD_z' assigned ");
        outfile.print(inParams.SD_z);
        outfile.println(" in module InputFormat");
        outfile.close();
        infile.close();
    }
}
